src/Peering/Peering/Direct/NewAzureDirectPeeringConnectionCommand.cs

```diff
@@ -0,0 +1,204 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.Peering.Direct
+{
+    using System;
+    using System.Management.Automation;
+
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Common;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Models;
+
+    /// <summary>
+    /// NewAzureDirectPeeringConnectionCommand
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.New,
+        "AzPeeringDirectConnectionObject",
+        DefaultParameterSetName = Constants.ParameterSetNameIPv4Prefix, SupportsShouldProcess = true)]
+    [OutputType(typeof(PSDirectConnection))]
+    public class NewAzureDirectPeeringConnectionCommand : PeeringBaseCmdlet
+    {
+        /// <summary>
+        /// Gets or sets the InputObject Facility DB.
+        /// </summary>
+        [Parameter(
+        Position = Constants.PositionPeeringZero,
+        Mandatory = true,
+        ValueFromPipelineByPropertyName = true,
```

Comment:
ValueFromPipelineByPropertyName should only be applied to ResourceId parameters.